core/vdbe/insn.rs

```diff
@@ -845,3 +850,16 @@ fn compute_shr(lhs: i64, rhs: i64) -> i64 {
         lhs >> rhs
     }
 }
+
+pub fn exec_boolean_not(mut reg: &OwnedValue) -> OwnedValue {
+    if let OwnedValue::Agg(agg) = reg {
+        reg = agg.final_value();
+    }
+    match reg {
+        OwnedValue::Null => OwnedValue::Null,
+        OwnedValue::Integer(i) => OwnedValue::Integer((*i == 0) as i64),
+        OwnedValue::Float(f) => OwnedValue::Integer((*f == 0.0) as i64),
+        OwnedValue::Text(text) => exec_boolean_not(&cast_text_to_numerical(&text.value)),
+        _ => todo!(),
```

Comment:
Can this be a panic! instead with an error message that this should not happen on `other_type`

Reply:
Agreed, I just kept the same logic from other functions like `exec_bit_not` . 

We should probably fix it across all these functions. Will try to make a PR for this.

Reply:
@jussisaurio @psvri We're a library, perhaps it's better to never panic, actually.